src/answers.sql

```diff
@@ -1,15 +1,250 @@
 -- Your answers here:
 -- 1
+--
+SELECT
+	TYPE ACCOUNT_TYPE,
+	ROUND(SUM(MOUNT)::NUMERIC, 2) MONEY_SUM
+FROM
+	ACCOUNTS
+GROUP BY
+	TYPE;
 
+--
+--
+--
+--
 -- 2
+--
+SELECT
+	COUNT(*) AS COUNT_USER_AT_LEAST_2_CURRENT_ACCOUNT
+FROM
+	(
+		SELECT
+			U.ID
+		FROM
+			USERS U
+			INNER JOIN ACCOUNTS A ON U.ID = A.USER_ID
+		WHERE
+			A.TYPE = 'CURRENT_ACCOUNT'
+		GROUP BY
+			U.ID
+		HAVING
+			COUNT(U.ID) > 1
+	);
 
+--
+--
+--
+--
+--
 -- 3
+--
+SELECT
+	ID,
+	ACCOUNT_ID,
+	TYPE,
+	MOUNT
+FROM
+	ACCOUNTS
+ORDER BY
+	MOUNT DESC
+LIMIT
+	5;
 
+--
+--
+--
+--
 -- 4
+-- I MAKE SURE I USE EVERY ACCOUNT BY UNION ALL!
+-- INNER JOIN MAY LEAVE OUT USERS WITH NO MOVEMENTS
+CREATE
+OR REPLACE FUNCTION GET_ACCOUNTS_FINAL_BALANCE () RETURNS TABLE (ACCOUNT_ID UUID, MONEY NUMERIC(15, 2)) AS $$ BEGIN RETURN QUERY WITH accounts_money_movements AS (
+        (
+            -- account_from transactions
+            SELECT account_from AS account_id,
+                CASE
+                    WHEN type = 'IN' THEN mount
+                    ELSE mount * -1
+                END AS mount
+            FROM movements
+        )
+        UNION ALL
```

Comment:
Why using `union all` instead of `union`

Reply:
Asumme all movement incoming to an account is "1000", and starting mount on the same account is "1000".

UNION would take both rows as the same, while in reality the account should have 2000 as total balance!

This way I make sure to have all movements related to the account (account.from + account.to + account.startingMount) = account.finalBalance

Reply:
That's a pretty good explanation, with a nice example! Good job!